File: src/ClassMaker/Form1.vb
Imports System.Reflection

Public Class Form1
    Public Enum Language
        VBNet = 0
        CSharp = 1
        Python = 2
        Php = 3
    End Enum


    Public Function WriteImports(ByVal lang As Language) As String
        Dim import As String = Nothing
        Select Case lang
            Case Language.VBNet
                import = "Imports System.Collections.Generic" & vbCrLf & _
                       "Imports System.Drawing" & vbCrLf & _
                       "Imports System.IO" & vbCrLf & vbCrLf

            Case Language.CSharp
                import = "using System.Collections.Generic;" & vbCrLf & _
                       "using System.Drawing;" & vbCrLf & _
                       "using System.IO;" & vbCrLf & vbCrLf

            Case Language.Python
                import = "import struct" & vbCrLf & _
                    "import getopt" & vbCrLf & _
                    "import sys" & vbCrLf & _
                    "import os" & vbCrLf & vbCrLf

            Case Language.Php
                import = Nothing

        End Select
        Return import
    End Function

    Public Function WriteHeader(ByVal thislang As Language, ByVal isNamespace As Boolean) As String
        Dim header As String = Nothing

        Select Case thislang
            Case Language.VBNet
                If isNamespace Then
                    header = "Namespace " & TextBox6.Text & vbCrLf
                End If
                header &= vbTab & "Public Class " & TextBox5.Text & vbCrLf
                If CheckBox1.Checked = True Then
                    header &= vbTab & vbTab & vbTab & "Inherits " & TextBox4.Text & vbCrLf & vbCrLf
                End If

            Case Language.CSharp
                If isNamespace Then
                    header = "Namespace " & TextBox6.Text & "{" & vbCrLf
                End If
                header &= vbTab & "public Class " & TextBox5.Text
                If CheckBox1.Checked = True Then
                    header &= " : " & TextBox4.Text
                End If
                header &= " {" & vbCrLf & vbCrLf


            Case Language.Python
                header = "class " & TextBox5.Text
                If CheckBox1.Checked = True Then
                    header &= "(" & TextBox4.Text & ")" & vbCrLf
                Else
                    header &= "(object):" & vbCrLf
                End If


            Case Language.Php
                header = Nothing

        End Select
        Return header
    End Function

    Public Function WriteFields(ByVal thisLang As Language, ByVal thisFields() As String) As String
        Dim fields As String = Nothing


        For i As Integer = 0 To UBound(thisFields)
            Dim thisType As String = "String"
            Dim strField() As String = Split(thisFields(i), "#")
            Dim varName As String = thisFields(i)

            If Not strField Is Nothing Then
                thisType = strField(1)
                varName = strField(0)
            End If
            Select Case thisLang
                Case Language.VBNet
                    fields &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

                Case Language.CSharp
                    fields &= vbTab & vbTab & "Private " & thisType & " me" & varName & ";" & vbCrLf

                Case Language.Python
                    'todo   fields &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

                Case Language.Php
                    fields = Nothing

            End Select
        Next

        Return fields & vbCrLf
    End Function

    Public Function WriteProperties(ByVal thisLang As Language, ByVal thisFields() As String) As String
        Dim Properties As String = Nothing


        For i As Integer = 0 To UBound(thisFields)
            Dim thisType As String = "String"
            Dim strField() As String = Split(thisFields(i), "#")
            Dim varName As String = thisFields(i)

            If Not strField Is Nothing Then
                thisType = strField(1)
                varName = strField(0)
            End If
            Select Case thisLang
                Case Language.VBNet
                    Properties &= vbTab & vbTab & "Public Property " & varName & "() as " & thisType & vbCrLf & _
                            vbTab & vbTab & vbTab & "Get" & vbCrLf & _
                            vbTab & vbTab & vbTab & vbTab & "Return me" & varName & vbCrLf & _
                            vbTab & vbTab & vbTab & "End Get" & vbCrLf & _
                            vbTab & vbTab & vbTab & "Set(value as " & thisType & ")" & vbCrLf & _
                            vbTab & vbTab & vbTab & vbTab & "me" & varName & " = value" & vbCrLf & _
                            vbTab & vbTab & vbTab & "End Set" & vbCrLf & _
                            vbTab & vbTab & "End Property" & vbCrLf & vbCrLf

                Case Language.CSharp
                    thisType = Replace(thisType, "Boolean", "bool")
                    thisType = Replace(thisType, "Single", "float")
                    thisType = Replace(thisType, "Object", "var")
                    Properties &= vbTab & vbTab & "public " & thisType & " " & varName & " {" & vbCrLf & _
                        vbTab & vbTab & vbTab & "get { return this.me" & varName & "; }" & vbCrLf & _
                        vbTab & vbTab & vbTab & "set { this.me" & varName & " = value; }" & vbCrLf & _
                        vbTab & vbTab & "}" & vbCrLf & vbCrLf

                Case Language.Python
                    'todo   fields &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

                Case Language.Php
                    Properties = Nothing

            End Select
        Next

        Return Properties
    End Function

    Public Function WriteFooter(ByVal thisLang As Language, ByVal isNamespace As Boolean) As String
        Dim footer As String = Nothing

        Select Case thisLang
            Case Language.VBNet
                If isNamespace Then
                    footer &= vbTab & vbTab & "End Class" & vbCrLf
                    footer &= vbTab & "End Namespace" & vbCrLf
                Else
                    footer &= vbTab & "End Class" & vbCrLf
                End If

            Case Language.CSharp
                If isNamespace Then
                    footer &= vbTab & vbTab & "}" & vbCrLf
                End If
                footer &= vbTab & "}" & vbCrLf

            Case Language.Python
                'todo   footer &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

            Case Language.Php
                footer = Nothing

        End Select


        Return footer
    End Function

    Public Function WriteCreators(ByVal thisLang As Language, ByVal isNamespace As Boolean, ByVal thisFields() As String) As String
        Dim Creators As String = Nothing

        Dim header As String = Nothing
        Dim content As String = Nothing
        Dim footer As String = Nothing



        For i As Integer = 0 To UBound(thisFields)
            Select Case thisLang
                Case Language.VBNet
                    If isNamespace Then
                    End If
                    header &= vbTab & vbTab & "Public Sub New("

                Case Language.CSharp
                    If isNamespace Then
                    End If
                    header &= vbTab & vbTab & "public " & TextBox5.Text & "("

                Case Language.Python
                    header = vbTab & "def __init__(self,"

                Case Language.Php


            End Select
            Dim strType() As String = Split(thisFields(i), ";")
            'todo
            For j As Integer = 0 To UBound(strType)
                Dim strType2() As String = Split(strType(j), "#")
                Dim varName As String = strType2(0)
                Dim varType As String = strType2(1)
                If j > 0 And j <= UBound(strType) Then
                    header &= ", "
                End If
                Select Case thisLang
                    Case Language.VBNet
                        If isNamespace Then
                        End If
                        header &= "Byval this" & varName & " as " & varType
                        content &= vbTab & vbTab & vbTab & "me" & varName & " = this" & varName & vbCrLf
                        If j >= UBound(strType) Then
                            header &= ")" & vbCrLf
                           
                        End If

                    Case Language.CSharp
                        If isNamespace Then
                        End If

                        header &= varType & " this" & varName
                        content &= vbTab & vbTab & vbTab & "this.me" & varName & " = this" & varName & ";" & vbCrLf
                        If j >= UBound(strType) Then
                            header &= ")"
                            If CheckBox2.Checked = True Then
                                header &= ":" & TextBox4.Text & " {" & vbCrLf
                            Else
                                header &= " {" & vbCrLf
                            End If
                        End If

                    Case Language.Python
                        header &= "this" & varName ', ThisDug, thisEntrance):"
                        content &= vbTab & vbTab & "self._me" & varName & " = this" & varName & vbCrLf
                        If j >= UBound(strType) Then
                            header &= "):" & vbCrLf
                        End If

                    Case Language.Php


                End Select

            Next

            Select Case thisLang
                Case Language.VBNet
                    If isNamespace Then
                    End If
                    footer = vbTab & vbTab & "End Sub" & vbCrLf & vbCrLf

                Case Language.CSharp
                    If isNamespace Then
                    End If
                    footer = vbTab & vbTab & "}" & vbCrLf & vbCrLf

                Case Language.Python
                    'todo   footer &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

                Case Language.Php
                    footer = Nothing

            End Select
            Creators &= header & content & footer
        Next
        Return Creators
    End Function

    Private Sub Button1_Click(sender As System.Object, e As System.EventArgs) Handles Button1.Click, Button3.Click, Button2.Click
        Dim str1() As String = Split(Replace(TextBox1.Text, vbTab, "#"), vbCrLf)
        Dim str2() As String = Split(Replace(TextBox3.Text, vbTab, "#"), vbCrLf)
        Dim lang As Language = ComboBox1.SelectedIndex
        TextBox2.Text = WriteImports(lang) & WriteHeader(lang, CheckBox5.Checked) & WriteFields(lang, str1) & WriteCreators(lang, CheckBox5.Checked, str2) & _
            WriteProperties(lang, str1) & WriteFooter(lang, CheckBox5.Checked)

    End Sub

  

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        ComboBox1.SelectedIndex = 0

    End Sub

    
End Class

File: src/ClassMaker/Form1.cs
using System;
using System.Linq;
using System.Windows.Forms;

namespace ClassMakerCSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

		public enum Language
		{
			VBNet = 0,
			CSharp = 1,
			Python = 2,
			Php = 3
		}

		public string WriteImports(Language lang)
		{
			string import = null;
			switch (lang) {
				case Language.VBNet:
                    import = "Imports System.Collections.Generic\r\nImports System.Drawing\r\nImports System.IO\r\n\r\n";

					break;
				case Language.CSharp:
                    import = "using System.Collections.Generic;\r\nusing System.Drawing;\r\nusing System.IO;\r\n\r\n";

					break;
				case Language.Python:
                    import = "import struct\r\nimport getopt\r\nimport sys\r\nimport os\r\n\r\n";

					break;
				case Language.Php:
					import = null;

					break;
			}
			return import;
		}

		public string WriteHeader(Language thislang, bool isNamespace)
		{
			string header = null;

			switch (thislang) {
				case Language.VBNet:
					if (isNamespace) {
						header = "Namespace " + TextBox6.Text + "\r\n";
					}
					header += "	" + "Public Class " + TextBox5.Text + "\r\n";
					if (CheckBox1.Checked == true) {
						header += "	" + "	" + "	" + "Inherits " + TextBox4.Text + "\r\n" + "\r\n";
					}

					break;
				case Language.CSharp:
					if (isNamespace) {
						header = "Namespace " + TextBox6.Text + "{" + "\r\n";
					}
					header += "	" + "public Class " + TextBox5.Text;
					if (CheckBox1.Checked == true) {
						header += " : " + TextBox4.Text;
					}
					header += " {\n\n";

					break;

				case Language.Python:
					header = "class " + TextBox5.Text;
					if (CheckBox1.Checked == true) {
						header += "(" + TextBox4.Text + ")" + "\r\n";
					} else {
						header += "(object):" + "\r\n";
					}

					break;

				case Language.Php:
					header = null;

					break;
			}
			return header;
		}

		public string WriteFields(Language thisLang, string[] thisFields)
		{
			string fields = null;


			for (int i = 0; i <= thisFields.Length-1 ; i++) {
				string thisType = "String";
				string[] strField = thisFields[i].Split('#');
				string varName = thisFields[i];

				if ((strField != null)) {
					thisType = strField[1];
					varName = strField[0];
				}
				switch (thisLang) {
					case Language.VBNet:
						fields += "	" + "	" + "Private me" + varName + " as " + thisType + "\r\n";

						break;
					case Language.CSharp:
						fields += "	" + "	" + "Private " + thisType + " me" + varName + ";" + "\r\n";

						break;
					case Language.Python:
						break;
					//todo   fields &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

					case Language.Php:
						fields = null;

						break;
				}
			}

			return fields + "\r\n";
		}

		public string WriteProperties(Language thisLang, string[] thisFields)
		{
			string properties = null;


			for (int i = 0; i <= thisFields.Length-1; i++) {
				string thisType = "String";
				string[] strField = thisFields[i].Split( '#');
				string varName = thisFields[i];

				if ((strField != null)) {
					thisType = strField[1];
					varName = strField[0];
				}
				switch (thisLang) {
					case Language.VBNet:
						properties += "\t" + "\t" + "Public Property " + varName + "() as " + thisType + "\r\n" + "	" + "	" + "	" + "Get" + "\r\n" + "	" + "	" + "	" + "	" + "Return me" + varName + "\r\n" + "	" + "	" + "	" + "End Get" + "\r\n" + "	" + "	" + "	" + "Set(value as " + thisType + ")" + "\r\n" + "	" + "	" + "	" + "	" + "me" + varName + " = value" + "\r\n" + "	" + "	" + "	" + "End Set" + "\r\n" + "	" + "	" + "End Property\r\n\r\n";

						break;
					case Language.CSharp:
                        thisType = thisType.Replace( "Boolean", "bool");
                        thisType = thisType.Replace( "Single", "float");
                        thisType = thisType.Replace( "Object", "var");
						properties += "\t" + "\t" + "public " + thisType + " " + varName + " {" + "\r\n" + "	" + "	" + "	" + "get { return this.me" + varName + "; }" + "\r\n" + "	" + "	" + "	" + "set { this.me" + varName + " = value; }" + "\r\n" + "	" + "	" + "}\r\n\r\n";

						break;
					case Language.Python:
						break;
					//todo   fields &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

					case Language.Php:
						properties = null;

						break;
				}
			}

			return properties;
		}

		public string WriteFooter(Language thisLang, bool isNamespace)
		{
			string footer = null;

			switch (thisLang) {
				case Language.VBNet:
					if (isNamespace) {
						footer += "\t" + "\t" + "End Class" + "\r\n";
						footer += "\t" + "End Namespace" + "\r\n";
					} else {
						footer += "\t" + "End Class" + "\r\n";
					}

					break;
				case Language.CSharp:
					if (isNamespace) {
						footer += "\t" + "\t" + "}" + "\r\n";
					}
					footer += "\t" + "}" + "\r\n";

					break;
				case Language.Python:
					break;
				//todo   footer &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

				case Language.Php:
					footer = null;

					break;
			}


			return footer;
		}

		public string WriteCreators(Language thisLang, bool isNamespace, string[] thisFields)
		{
			string creators = null;

			string header = null;
			string content = null;
			string footer = null;

            for (int i = 0; i <= thisFields.Length-1; i++)
            {
				switch (thisLang) {
					case Language.VBNet:
						if (isNamespace) {
						}
                        header += "\t\tPublic Sub New(";

						break;
					case Language.CSharp:
						if (isNamespace) {
						}
                        header += "\t\tpublic " + TextBox5.Text + "(";

						break;
					case Language.Python:
                        header = "\tdef __init__(self,";

						break;
					case Language.Php:

						break;

				}
				string[] strType = thisFields[i].Split(';');
				//todo
				for (int j = 0; j <= strType.Length-1; j++) {
					string[] strType2 = strType[j].Split('#');
					string varName = strType2[0];
					string varType = strType2[1];
                    if (j > 0 & j <= strType.Length-1)
                    {
						header += ", ";
					}
					switch (thisLang) {
						case Language.VBNet:
							if (isNamespace) {
							}
							header += "Byval this" + varName + " as " + varType;
                            content += "\t\t\tme" + varName + " = this" + varName + "\r\n";
                            if (j >= strType.Length-1)
                            {
								header += ")" + "\r\n";

							}

							break;
						case Language.CSharp:
							if (isNamespace) {
							}

							header += varType + " this" + varName;
                            content += "\t\t\tthis.me" + varName + " = this" + varName + ";" + "\r\n";
							if (j >= strType.Length) {
								header += ")";
								if (CheckBox2.Checked == true) {
									header += ":" + TextBox4.Text + " {" + "\r\n";
								} else {
									header += " {" + "\r\n";
								}
							}

							break;
						case Language.Python:
							header += "this" + varName;
							//, ThisDug, thisEntrance):"
                            content += "\t\tself._me" + varName + " = this" + varName + "\r\n";
							if (j >= strType.Length-1) {
								header += "):" + "\r\n";
							}

							break;
						case Language.Php:

							break;

					}

				}

				switch (thisLang) {
					case Language.VBNet:
						if (isNamespace) {
						}
                        footer = "\t" + "\t" + "End Sub\r\n\r\n";

						break;
					case Language.CSharp:
						if (isNamespace) {
						}
                        footer = "\t" + "\t" + "}\r\n\r\n";

						break;
					case Language.Python:
						break;
					//todo   footer &= vbTab & vbTab & "Private me" & varName & " as " & thisType & vbCrLf

					case Language.Php:
						footer = null;

						break;
				}
				creators += header + content + footer;
			}
			return creators;
		}

		private void Button1_Click(System.Object sender, System.EventArgs e)
		{
            string str = TextBox1.Text.Replace('\t', '#');
			string[] str1 = str.Split('\n');
            string strb = TextBox3.Text.Replace('\t', '#');
			string[] str2 = strb.Split('\n');
            Language lang = (Language) ComboBox1.SelectedIndex;
			TextBox2.Text = WriteImports(lang) + WriteHeader(lang, CheckBox5.Checked) + WriteFields(lang, str1) + WriteCreators(lang, CheckBox5.Checked, str2) + WriteProperties(lang, str1) + WriteFooter(lang, CheckBox5.Checked);

		}



		private void Form1_Load(System.Object sender, System.EventArgs e)
		{
			ComboBox1.SelectedIndex = 0;

		}

        private void Label6_Click(object sender, System.EventArgs e)
        {

        }

        private void CheckBox1_CheckedChanged(object sender, System.EventArgs e)
        {

        }

      

       


	}
}
